packages/office-addin-dev-settings/src/cli.ts

```diff
@@ -62,6 +62,12 @@ commander
     .option("-a,--app <app>", `The Office app to launch. ("Excel", "PowerPoint", or "Word")`)
     .action(commands.sideload);
 
+commander
+    .command("sideload-web <document-url> <manifest-path> <port>")
```

Comment:
Why a separate command? Let's try to make desktop and web sideload more unified.

Reply:
![image](https://user-images.githubusercontent.com/741153/94471196-98bd1380-017d-11eb-9a3c-fcbd7377d8a0.png)

Seems like we can have only one sideload command, but add parameters to allow for specifying whether desktop or web, and then have optional parameters that can be added which apply for web.

Reply:
The startDebugging() command already has AppType enum which will be defined as AppType.Desktop or AppType.Web. If the sideloadCommand has an optional AppType parameter added which can be used to specify whether to sideload to desktop or web. It can default to AppType.Desktop. 

For the implementation, the registration of the add-in can be skipped when appType is not Desktop.

There is just a small amount of logic from that point to generate the sideload file/url and then open it.

Reply:
It would be nice to augment desktop sideload to have an optional parameter to specify an existing file. It can validate that the file already has the add-in webextension id/version in it or error out if not.

Thus, the --document (or --doc if we want to shorten it) can then work for both desktop and web.

Sideload to web would need to have the --document parameter (at least for now). It still leaves open implementing a way to create a new doc or looking for one so we could relax the need to specify the existing document.

Reply:
I thought of having a unified command as well initially but couldn't come up with a clean solution. I will revisit however

Reply:
I think the problem with having a doc param that can apply for desktop as well as web is that the doc must be of a specific type - it has to have a webextension xml part already present in it for our current sideloading code work or else we fail here:

  // replace the placeholder id and version
  const zipFile = zip.file(webExtensionPath);
  if (!zipFile) {
    throw new Error("webextension was not found.")
  }

I think having a doc param that applies for both web and desktop will confuse users and they will specify just any document that doesn't have a webextenion part in it already and the operation will fail.  

We could of course try rewriting our desktop sideloading code to try and accomodate this scenario, but I am not sure how difficult that would be at this point

Reply:
Yeah, it would be beneficial to be able to have the desktop sideload support that. There is an issue asking for it. Obviously the logic becomes different if we are creating the document versus checking that it already exists and using it. This should not be difficult to achieve.

Reply:
```
  await registerAddIn(manifestPath);

  if (app) {
    const sideloadFile = await generateSideloadFile(app, manifest);

    await open(sideloadFile, { wait: false });
  }
```

In this code, if the sideloadFile is provided, then use it rather than generate the file.

Reply:
I suppose we can just create a new webExtensionXml part if it doesn't already exist in the document specified via the doc paramter. Probably not too difficult